src/main/java/nyx/Deadline.java

```diff
@@ -0,0 +1,53 @@
+package nyx;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Represents a task with a deadline.
+ * This class inherits from the parent abstract Task class.
+ */
+public class Deadline extends Task {
+    private static final String DATETIME_FORMAT = "yyyy-MM-dd H:m";
+    private final LocalDateTime by;
```

Comment:
perhaps a better name for the variable might be `endDateTime`. it is not immediately obvious what `by` refers to!